### PR TITLE
chore: backport tx status

### PR DESCRIPTION
**Pull request by ninabarbakadze**
## Description

backport TxStatus work from main

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -445,6 +450,35 @@ func (bs *BlockStore) SaveSeenCommit(height int64, seenCommit *types.Commit) err
 	return bs.db.Set(calcSeenCommitKey(height), seenCommitBytes)
 }
 
+// SaveTxInfo indexes the txs from the block with the given response codes from execution.
+func (bs *BlockStore) SaveTxInfo(block *types.Block, txResponseCodes []uint32) error {
```
Comment (ninabarbakadze) [inline store/store.go:454]: should i move this to executor since that's where it's called now? @cmwaters

**Review by cmwaters — APPROVED** [review]
LGTM

File: rpc/client/http/http.go [hunk 2]
```diff
@@ -516,6 +516,24 @@ func (c *baseRPCClient) DataCommitment(
...
+		"hash": hash,
+	}
+
+	_, err := c.caller.Call(ctx, "tx_status", params, result)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+
```
Comment (rach-id) [inline rpc/client/http/http.go:534]:
```suggestion
```

**Review by rach-id — COMMENTED** [review]
first pass looks good to me 👍 
Is there an issue for this or more context?

**Comment by ninabarbakadze** [comment]
> first pass looks good to me 👍 Is there an issue for this or more context?

Thanks for the review! Here are the links to the work I'm backporting from main: 
- https://github.com/celestiaorg/celestia-core/issues/1281
- https://github.com/celestiaorg/celestia-core/pull/1287
- https://github.com/celestiaorg/celestia-core/issues/1397

File: rpc/core/tx.go [hunk 3]
```diff
@@ -17,6 +17,13 @@ import (
 	"github.com/tendermint/tendermint/types"
 )
 
+const (
+	txStatusUnknown   string = "UNKNOWN"
```
Comment (rach-id) [inline rpc/core/tx.go:21]: [non-blocking][microscopic nit]
probably document the cases where the status is unknown

File: rpc/core/types/responses.go [hunk 4]
```diff
@@ -60,6 +60,15 @@ type ResultCommit struct {
 	CanonicalCommit    bool `json:"canonical"`
 }
 
+// ResultTxStatus represents the status of a transaction during its life cycle.
+// It contains info to locate a tx in a committed block as well as its execution code and status.
+type ResultTxStatus struct {
+	Height        int64  `json:"height"`
+	Index         uint32 `json:"index"`
+	ExecutionCode uint32 `json:"execution_code"`
+	Status        string `json:"status"`
```
Comment (rach-id) [inline rpc/core/types/responses.go:69]: [non blocking]
maybe document these

**Comment by ninabarbakadze** [comment]
> utAck.
> 
> Awesome stuff 🔥 Left some unblocking nits.
> 
> Also, would be good to update the openapi specs to reflect these changes both in main in this branch. Can be handled in a separate PR or an issue for this to be done later.

Opened an issue https://github.com/celestiaorg/celestia-core/issues/1406

**Review by cmwaters — APPROVED** [review]

**Review by rootulp — APPROVED** [review]

**Pull request merged**